packages/api-core/src/resources/organizations.js

```diff
@@ -99,47 +109,121 @@ export default class AvOrganizations extends AvApi {
+          (matchedANDOrgsByPerm, permissionIdAND, index) => {
+            this.userPermissions[`${permissionIdAND}`].organizations.forEach(
+              org => {
+                if (index === 0) {
+                  // add the orgs for the first permission
+                  matchedANDOrgsByPerm[org.id] = org;
+                } else if (matchedANDOrgsByPerm[org.id]) {
+                  // if duplicate, add resources
+                  matchedANDOrgsByPerm[org.id].resources = matchedANDOrgsByPerm[
+                    org.id
+                  ].resources.concat(org.resources);
+                }
+              }
+            );
+            // filter unmatched orgs out
+            matchedANDOrgsByPerm = Object.keys(matchedANDOrgsByPerm)
+              .filter(orgId =>
+                this.userPermissions[permissionIdAND].organizations.some(
+                  org => org.id === orgId
+                )
+              )
+              .reduce((obj, orgId) => {
+                obj[orgId] = matchedANDOrgsByPerm[orgId];
+                return obj;
+              }, {});
+
+            return matchedANDOrgsByPerm;
+          },
+          {}
+        );
+        Object.keys(matchedOrgs).forEach(orgId => {
+          if (!accum[orgId]) {
+            accum[orgId] = matchedOrgs[orgId];
+            accum[orgId].match = false;
+          }
+        });
+      } else {
+        // just one permission, get the orgs under this permission
+        this.userPermissions[permissionIdOR].organizations.forEach(org => {
+          if (!accum[org.id]) {
+            accum[org.id] = org;
+            accum[org.id].match = false;
+          } else {
+            // add the resources
+            accum[org.id].resources = accum[org.id].resources.concat(
+              org.resources
+            );
           }
         });
+      }
+      return accum;
+    }, {});
 
-        return accum;
-      },
-      []
-    );
+    // loop thru the orgs from permission filtering and check resourceIds list to further filter
+    if (resourceIdsArray.length === 0) {
+      Object.keys(authorizedOrgs).forEach(orgId => {
+        authorizedOrgs[orgId].match = true;
+      });
+    } else {
+      resourceIdsArray.forEach(resourceIdOR => {
+        if (Array.isArray(resourceIdOR)) {
+          // there is AND logic
+          Object.keys(authorizedOrgs).forEach(orgId => {
+            if (authorizedOrgs[orgId]) {
+              const isMatch = resourceIdOR.every(resId =>
+                authorizedOrgs[orgId].resources.some(res => res.id === resId)
+              );
+              if (isMatch) {
+                authorizedOrgs[orgId].match = true;
+              }
+            }
+          });
+        } else {
+          Object.keys(authorizedOrgs).forEach(orgId => {
+            const isMatch = authorizedOrgs[orgId].resources.some(
+              res => res.id === resourceIdOR
+            );
+            if (isMatch || !resourceIdOR) {
```

Comment:
I think `|| !resourceIdOR` can be removed, `!"string"` will always be false

Reply:
it is checking for a blank string ""